staging/src/k8s.io/api/admissionregistration/v1beta1/types.go

```diff
@@ -265,6 +265,11 @@ type WebhookClientConfig struct {
 	// the webhook's server certificate.
 	// Required.
 	CABundle []byte `json:"caBundle" protobuf:"bytes,2,opt,name=caBundle"`
+
+	// `timeout` is the number of seconds after which the webhook request times out.
+	// Must be greater than 0 if set. If not set, will use server default 30s.
```

Comment:
I also remember 30s is the default, but I can't find it. Could you double check the code?

Reply:
It's here https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/authentication.go#L170~L175

Reply:
Does that code path actually get called when constructing admission webhook clients?

Reply:
Yes, this can be verified with your e2e case. IIRC, should be the hang case.